apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/repository/PermissionRepository.java

```diff
@@ -50,4 +50,8 @@ List<Permission> findByPermissionTypeInAndTargetId(Collection<String> permission
   @Modifying
   @Query("UPDATE Permission SET IsDeleted = true, DeletedAt = ROUND(UNIX_TIMESTAMP(NOW(4))*1000), DataChange_LastModifiedBy = ?2 WHERE Id in ?1 and IsDeleted = false")
   Integer batchDelete(List<Long> permissionIds, String operator);
+
+  @Query("SELECT p.id from Permission p where p.targetId = CONCAT(?1, '+', ?2, '+', ?3)"
+  + " AND ( p.permissionType = 'ModifyNamespacesInCluster' OR p.permissionType = 'ReleaseNamespacesInCluster')")
```

Comment:
Adding a `permissionType` check here is a good idea, as it may conflict with `ModifyNamespace` and `ReleaseNamespaces` permissions. It would also be wise to include the check in `List<Long> findPermissionIdsByAppIdAndNamespace(String appId, String namespaceName);`.